### ISSUE TITLE
U-Boot can't find storage device

### ISSUE THREAD

**Issue by emilia-miki** [tpwrules/nixos-apple-silicon#164]
I have a Macbook Air M1. After following the building stages and doing `dd if=nixos-*.iso of=/dev/sda`, installing UEFI and rebooting, U-Boot finds 1 USB device, then writes `usb_new_device: Cannot read configuration, skipping device XXXX:XXXX` and does not find any storage devices. Running `setenv usb_pgood_delay 1; usb reset` I encounter the same issue. Using a prebuilt ISO or a larger delay didn't help either. The USB device doesn't use any hubs and is directly connected to one of the 2 ports, changing ports didn't help either.

If I need to buy a new USB device, which is better: lightning or USB-C?

**Comment by emilia-miki** [comment]
Update: fixed by trying usb reset with default delay many times. The I encountered /dev/root mounting issue, which was fixed using instructions from another closed issue in this repo. Probably just a low quality USB device.

**Comment by tpwrules** [comment]
Sounds like it.

**Comment by maxlang7** [comment]
@tpwrules I'm having a vary similar issue, except my U-Boot gives a slightly different output. It scans for the USB devices and finds the usb I flashed with the most recent release iso, then says: the following output "Scanning nvme 0:4.... Failed to load '/' Loading Boot0000 'nvme 0:4' failed EFI boot manager: cannot load any image" (I think this part is fine, obviously it shouldn't try to boot from the internal ssd, but then it says: "Device 0: Vendor: USB Rev: 1.00 Prod: Sandisk 3.2Gen1 Type: Removable Hard Disk Capacity = 238560.0 MB = 232.9 GB ... is now current device. Scanning usb 0:1. Failed to load '/' Loading Boot0000 nvme 0:4 failed EFI boot manager: cannot load any image." 

If you've got time, I'd really appreciate any help. I just used dd to flash the iso onto the drive, but perhaps I'm using the wrong format or something.
